specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,522 @@
+leverage and combine the available instructions to express a program, but does
+not dictate which quantum instructions may be used. Targeting a program to a
+specific backend requires choosing a suitable profile and quantum instruction
+set (QIS). Both can be chosen largely independently, though certain instruction
+sets may be incompatible with this (or other) profile(s). The section on the
+[quantum instruction set](#quantum-instruction-set) defines the requirements for
+a QIS to be compatible with the Base Profile. More information about the role of
+the QIS, recommendations for front- and backend providers, as well as the
+distinction between runtime functions and quantum instructions can be found in
+[this document](../Instruction_Set.md).
+
+**Bullet 2: Measurements** <br/>
+
+The second requirement should be taken to mean that a Base Profile compliant
+program does *not* apply instructions to a qubit after it has been measured;
+instructions that result in a unitary transformation of the quantum state must
+be applied before performing any irreversible actions such as measurements. It
+specifically also implies the following:
+
+- There is no need for the quantum processor ([QPU](../Execution.md)) to be able
+to measure only a subset of all available qubits at a time.
+- Executing a Base Profile compliant program does not require support for
+applying quantum instructions dependent on measurement outcomes.
+
+**Bullet 3: Program output** <br />
+
+The specification of QIR and all its profiles needs to permit to accurately
+reflect the program intent. This includes being able to define and customize the
+program output. The Base Profile specification hence requires explicitly
+expressing which values/measurements are returned by the program and in which
+order. How to express this is defined in the section on [output
+recording](#output-recording).
+
+While it is sufficient for the QPU to do a final measurement of all qubits in a
+predefined order at the end of the program, only the selected subset will be
+reflected in the produced output schema. A suitable output schema can be
+generated in a post-processing step after the computation on the quantum
+processor itself has completed; customization of the program output hence does
+not require support on the QPU itself.
+
+The defined [output schemas](../output_schemas/) provide different options for
+how a backend may express the computed value(s). The exact schema can be freely
+chosen by the backend and is identified by a string label in the produced
+schema. Each output schema contains sufficient information to allow quantum
+programming frameworks to generate a user-friendly presentation of the returned
+values in the requested order, such as, e.g., a histogram of all results when
+running the program multiple times.
+
+## Program Structure
+
+A Base Profile compliant program is defined in the form of a single LLVM bitcode
```

Comment:
Worth noting, even if not important - we call out that a base profile program is a single module but then, towards the bottom, address the concerns of merging multiple modules.

I wonder if this suggests that Versioning should be addressed outside of the scope of any one profile.

Reply:
Yes, I think the question of versioning should be covered by the full spec. As it is, the full spec does not (yet) contain a paragraph about it. I intend to make a pass over the full spec once we have 2 profiles + the runtime initialization and arg handling spec'ed. I don't expect any changes for the base profile, but the phrasing may move into a different file that is then linked here instead of covered (only) here.